app/main.go

```diff
@@ -81,31 +125,25 @@ func sync(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	sd, err := stackdriver.NewAdapter(ctx)
+	sd, err := stackdriver.NewAdapter(ctx, *sdLookBackInterval)
 	if err != nil {
 		logAndReturnError(ctx, w, err)
 		return
 	}
 	defer sd.Close()
 
-	stats, err := tsbridge.NewCollector(ctx)
+	stats, err := tsbridge.NewCollector(ctx, *sdInternalMetricsProject)
 	if err != nil {
 		logAndReturnError(ctx, w, err)
 		return
 	}
 	defer stats.Close()
 
-	p, err := strconv.Atoi(os.Getenv("UPDATE_PARALLELISM"))
-	if err != nil {
-		logAndReturnError(ctx, w, fmt.Errorf("could not parse UPDATE_PARALLELISM: %v", err))
-		return
-	}
-	if p < 1 || p > 100 {
-		logAndReturnError(ctx, w, fmt.Errorf("expected UPDATE_PARALLELISM between 1 and 100; got %d", p))
-		return
+	if *updateParallelism < 1 || *updateParallelism > 100 {
+		logAndReturnError(ctx, w, fmt.Errorf("expected UPDATE_PARALLELISM between 1 and 100; got %d", *updateParallelism))
```

Comment:
Can you do this check in `main` so that the process won't start if the value is wrong?

As it is, it'll return OK to the health check but never be able to handle a query if this argument is wrong.

Reply:
Added a flag validator.

Reply:
Added a flag validator.